tests/units/units.cpp

```diff
@@ -2760,3 +2760,32 @@ TEST(Units, validateConnectionsWithBaseUnits)
 
     EXPECT_EQ(size_t(0), validator->issueCount());
 }
+
+TEST(Units, equivalentUnitsMatchingBuiltinUnits)
+{
+    libcellml::UnitsPtr units = libcellml::Units::create("kelvin");
+    units->addUnit(libcellml::Units::StandardUnit::KELVIN);
+
+    libcellml::UnitsPtr unitsClone = units->clone();
+
+    EXPECT_TRUE(libcellml::Units::compatible(units, unitsClone));
+}
+
+TEST(Units, equivalentUnitsMatchingBuiltinUnitsFromVariable)
+{
+    libcellml::ModelPtr model = libcellml::Model::create("unit_compatible_test");
+
+    libcellml::VariablePtr variable = libcellml::Variable::create("T");
+    libcellml::UnitsPtr units = libcellml::Units::create("kelvin");
+    units->addUnit(libcellml::Units::StandardUnit::KELVIN);
+
+    libcellml::UnitsPtr unitsClone = units->clone();
```

Comment:
Not needed for this test.

Reply:
[45e8f05](https://github.com/cellml/libcellml/pull/1145/commits/45e8f056efd9d4a6f4898d87fa005cf50caa3bf6)